### PR TITLE
feat(description-list): added columnar layout

### LINKED ISSUE #4281: Description List: provide an option to fill columns not rows first
The provided DescriptionListGroup items, will fill the page horizontally first, by filling the rows and then rows.

Would it be possible to provide a property to make them first fill in the columns?

In cockpit we use Description List in the following scenario:
Screen Shot 2020-09-10 at 16 10 48

As you can see some of the data are more related to each other, IOW there are groups.

Let's visualize the above data as two groups, group1 for CPU info and group2 for BIOS info.

Our React code of course has the groups in order,
```
<DescriptionListGroup>
item1-group1
item2-group1
item3-group1
</DescriptionListGroup>
<DescriptionListGroup>
item1-group2
item2-group2
item3-group2
</DescriptionListGroup>
```

Now the groups fill horizontal space. I feel like the users would expect to find the groups vertically listed though. I can achieve this effect by setting `grid-auto-flow: column` and also the `grid-template-rows` rule.

Resulting in:
Screen Shot 2020-09-10 at 16 15 50

I am just raising the point here, feel free to close the issue if you think the current implementation does not need this flexibility.

### PR DESCRIPTION
**Pull request by mattnolting**
closes #4281

### CODE REVIEW

File: src/patternfly/components/DescriptionList/examples/DescriptionList.md [hunk 1]
```diff
@@ -56,6 +56,21 @@ cssPrefix: pf-c-description-list
...
+```
+
+### Column fill, horizontal
+```hbs
+{{> description-list__example description-list--title="Column fill, horizontal" description-list--modifier="pf-m-fill-columns pf-m-horizontal"}}
+```
+
+### Column fill, horizontal, multi-column
+```hbs
+{{> description-list__example description-list--title="Column fill, multi-column" description-list--modifier="pf-m-fill-columns pf-m-horizontal pf-m-2-col pf-m-3-col-3"}}
```
Comment (mcoker) [inline src/patternfly/components/DescriptionList/examples/DescriptionList.md:71]: what does `pf-m-3-col-3` do?

**Review by mcoker — COMMENTED** [review]
Very nice!!! One comment about a class name.

File: src/patternfly/components/DescriptionList/examples/DescriptionList.md (hunk 1 above)
Comment (mattnolting) [reply src/patternfly/components/DescriptionList/examples/DescriptionList.md:71]: Ah! Doh.

Comment (mattnolting) [reply src/patternfly/components/DescriptionList/examples/DescriptionList.md]: Updated!

**Review by wise-king-sullyman — APPROVED** [review]

**Review by mcoker — APPROVED** [review]

**Pull request merged**